File: src/category_strings/q11_string_compression/StringCompression.js
const LogUtil = require('../../utils/LogUtil.js');

/**
 * Determine the smaller between compressed and original string
 * See README for details
 * @param {string} s - upper and lowercase (a-z) ASCII characters
 * @return {string} the smaller between the original string and the compressed string
 */
var stringCompression = function(s) {
    // Write your code here
    return s;
}

var test = function() {
    // Change these values to manually test your program
    let strInput = "abccddeeeee";
    let acceptedResponse = "a1b1c2d2e5";
    let actualResponse = stringCompression(strInput);
    new LogUtil().logResultsString("11: String Compression", acceptedResponse, actualResponse);
}

module.exports = test;

File: src/category_strings/q11_string_compression/StringCompression.py
from utils.LogUtil import LogUtil;

class StringCompression:

    # Determine the smaller between compressed and original string
    # See README for details
    # @param s - upper and lowercase (a-z) ASCII characters
    # @return - the smaller between the original string and the compressed string
    def string_compression(self, s: str) -> str:
        # Write your code here
        return s;

    def test(self):
        # Change these values to manually test your program
        str_input = "abccddeeeee";
        accepted_response = "a1b1c2d2e5"
        actual_response = self.string_compression(str_input);
        LogUtil().log_results_int("11: String Compression", accepted_response, actual_response);